x-pack/plugins/security_solution/public/actions/toggle_column/default/toggle_column.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { CellAction, CellActionExecutionContext } from '@kbn/cell-actions';
+
+import { fieldHasCellActions, isInSecurityApp } from '../../utils';
+import type { SecurityAppStore } from '../../../common/store';
+import { getScopedActions, isInTableScope, isTimelineScope } from '../../../helpers';
+import { timelineDefaults } from '../../../timelines/store/timeline/defaults';
+import { defaultColumnHeaderType, tableDefaults } from '../../../common/store/data_table/defaults';
+import { timelineSelectors } from '../../../timelines/store/timeline';
+import { dataTableSelectors } from '../../../common/store/data_table';
+import { DEFAULT_COLUMN_MIN_WIDTH } from '../../../timelines/components/timeline/body/constants';
+import { KibanaServices } from '../../../common/lib/kibana';
+
+export const ACTION_ID = 'security_toggleColumn';
+const ICON = 'listAdd';
+
+export interface ShowTopNActionContext extends CellActionExecutionContext {
+  metadata?: {
+    scopeId?: string;
+    isObjectArray?: boolean;
+  };
+}
+
+export const COLUMN_TOGGLE = i18n.translate(
+  'xpack.securitySolution.actions.toggleColumnToggle.label',
+  {
+    defaultMessage: 'Toggle column in table',
+  }
+);
+
+export const NESTED_COLUMN = (field: string) =>
+  i18n.translate('xpack.securitySolution.actions.toggleColumnToggle.nestedLabel', {
+    values: { field },
+    defaultMessage:
+      'The {field} field is an object, and is broken down into nested fields which can be added as columns',
+  });
+
+export const createToggleColumnAction = ({
+  store,
+  order,
+}: {
+  store: SecurityAppStore;
+  order?: number;
+}): CellAction<ShowTopNActionContext> => {
+  let currentAppId: string | undefined;
+
+  const { application: applicationService } = KibanaServices.get();
+
+  applicationService.currentAppId$.subscribe((appId) => {
+    currentAppId = appId;
+  });
```

Comment:
We actually don't need to do this in the Security CellActions anymore. The triggers to which we are registering them are local to Security.